modules/javafx.graphics/src/main/java/javafx/scene/CssStyleHelper.java

```diff
@@ -200,7 +200,9 @@ private static void updateParentTriggerStates(Styleable styleable, int depth, Ps
         // found when matching selectors
         Styleable parent = styleable.getStyleableParent();
         for(int n=1; n<depth; n++) {
-
+            if (parent == null) {
+                return;
+            }
```

Comment:
While this does prevent the exception, I think this is fixing a symptom rather than the root cause. The only reason that parent can ever be null here and in the other place you added the null check is that `Tooltip::getStyleableParent` is returning a different answer between the time it is called to calculate `depth` and the time it is used here in a loop. This is happening because `Tooltip::getStyleableParent` uses static state that is only valid when on the FX application thread.